prime-router/docs/proposals/0011-internal-formats/0011-internal-formats.md

```diff
@@ -0,0 +1,101 @@
+However, these object classes would not be restricted to only defined FHIR resources. The custom classes may, for example, adopt an HL7 model for ELR and FHIR model for other types. 
+
+## Analysis
+The proposal considers four criteria to evaluate each choice.  
+- How does the option support the new scenarios listed in the introduction? 
+- What choice will allow new people to come up to speed quickly? 
+- What constraints and guardrails do each option have to prevent future issues and bugs?
+- How much effort will it take to convert the existing pipeline to this option? 
+
+### New Use-Cases
+Since FHIR is a successor to HL7, it covers the scenarios that HL7 covers. Consequently, an FHIR-based implementation will support both the HL7 and FHIR use-cases, while an HL7-based implementation will not. 
+
+### New People
+FHIR resources are more familiar to a typical engineer since they use widely adopted mechanisms like REST and JSON. In addition, FHIR has better documentation and a wider open-source community. Finally, new engineers are more likely to have already invested time to learn FHIR. 
+
+### Guardrails
+FHIR has far more regular implementation for concepts such as organization and people, essential for public health information. In addition, FHIR discourages common but harmful HL7 practices like coding with exceptions and placing multiple values into a single field. 
+
+### Initial Effort
+The initial effort will be high for all options, essentially causing rewrites to all serializers, transforms, and filters. However, ReportStream will have more HL7 use-cases in the short term, so either a custom class or HL7 approach will be the easiest to get started. 
+
+### Summary
+The table below grades each option on a 1-5 scale, where five is best. 
+
+|            | New Use-Cases | New People | Guardrails  | Initial Effort  |
+|:----       |:----------:   |:----------:|:-----------:|:---------------:|
+| HL7        | 3             | 3          | 2           | 4               |
+| FHIR       | 5             | 4          | 5           | 2               |
+| Custom     | 4             | 2          | 1           | 5               |
+
+## Recommendation
+Although an HL7 internal format would be the quickest path to support full ELR, since ELR is an HL7-based standard, this path would not support the future FHIR scenarios. HL7 also has problems representing new data elements like those that SimpleReport uses to capture new testing sites. Although it may take more initial work, ReportStream should use FHIR as its internal structure because it supports more future scenarios and benefits engineers.  
+
+## Details
+
+### Topics
+In this proposal, the concept of a topic becomes something like parts of a FHIR implementation guide which defines what FHIR resources are in a particular bundle. Further investigation should be done to see if they are the same concept. 
```

Comment:
We definitely need _something_ to provide a clear description of what we expect from a sender - and right now that's our .schema files.  So, yes, let's examine whether FHIR profiles could replace our current schemas is well worth considering.   This can be treated as a separate follow-on project to the FHIR-in-the-hole project.

Despite being designed for FHIR, the FHIR profiles would have to "work" for CSVs too - CSVs are not going away.